pkg/downloader/manager.go

```diff
@@ -233,6 +233,26 @@ func (m *Manager) loadChartDir() (*chart.Chart, error) {
 // This returns a lock file, which has all of the dependencies normalized to a specific version.
 func (m *Manager) resolve(req []*chart.Dependency, repoNames map[string]string) (*chart.Lock, error) {
 	res := resolver.New(m.ChartPath, m.RepositoryCache)
+	// NOTE: When OCI dependencies specify a semver range in Version,
+	// (*Resolver).Resolve must somehow get the list of OCI tag versions
+	// available to check against this constraint. However for backward
+	// compatibility we can not change that function signature to pass a
+	// *registry.Client required to get tags from the ORAS tag listing API.
+	// However we can add a new backward compatible struct field to
+	// *chart.Dependency, so that we can pass these along to
+	// (*Resolver).resolve from here through reqs []*chart.Dependency.
+	for i, d := range req {
+		if registry.IsOCI(d.Repository) {
+			// TODO(scottrigby): fix HTTP error
+			// > Error: Get "https://localhost:5000/v2/helm-charts/tags/list": http: server gave HTTP response to HTTPS client
+			ref := strings.TrimPrefix(d.Repository, "oci://")
+			tags, err := m.RegistryClient.Tags(ref)
+			if err != nil {
+				return nil, err
+			}
+			req[i].OCITagVersions = tags
+		}
+	}
```

Comment:
Why was this additional loop added here? This could be done in `internal/resolver/resolver.go` at the same place where `OCITagVersions` is used. Then the `Dependency` struct doesn't need to have any new property put on it. Localizing the work to the resolver would make the change simpler with fewer possible complications to consider.

Reply:
The problem is I don't see a backward compatible way to be able to get `RegistryClient` from within (*Resolver).Resolve, which is needed to call the API. However the calling function `(*Manager).resolve` has this already. So making the tag API call from there, then attaching the result to `Dependency` object was a solution for this that is backward compatible. Do you see another way?

Reply:
The resolver is an [`internal`](https://go.dev/doc/go1.4#internalpackages) package and not part of the public API. Take a look at its location. This is not a general purpose resolver meant to be imported and used by others. Its only access is through the `resolve` method here. It's meant just to support Helms specific use.

You already added a 3rd argument to `resolver.New` to pass in the registry client and you extended `Resolver` to hold on to it. Now you just need to use it within the `Resolve` method.

Reply:
@mattfarina thats exactly the approach that I am working on today

Reply:
Thanks @mattfarina you're right – and yes that makes it easier! @sabre1041 & I talked about this approach last night. Yep I see Andy added that now. Much better

Reply:
In touch w Andy on this, so we will get it done tonight one way or another 💯